.travis.yml

```diff
@@ -8,24 +8,36 @@ env:
   - secure: lFak7oRDvrUpzXkYEEHzFnPQiJhqnKx01oINSQirg/4KXnENzHKe1laz9dxi1l/nR6GSLyKcjkvUNUEaF6DKY4/Sv0OnivWLba9L6BebRw3+SC3NfakZS9DbTZKFMFS4F7BM4nuOEeSDZxn7Uab5Gx7Z4O73IMOmxtBkdK+YUth3iZJ9VVXaJkdaPurPmoKRxlg+ehIirwX6TX/ewpq1soOPH+CZ3HMxd6RRpxj0hp12Vwevh6/uDFJH4c34RG6SIR1kGN/r/gcq4vrkdz35YjbdhU0K52JQc27D8P5yhhP/THDh/EZS6Kxk1ou10P0yfm3hWNpp2Z+aO4wuokLTbVcygLxA+nMRy6/GioOOKM6e0iINjnz/c/RlCgImQeaWgHuXTjib0/M5tG3OpltDA5eDRhpGcarS26fzq4Sb8j3cP5C5MFuhj4k+rE0EQlzQqjnLEo8Wzo1f4KnYFW4SIRrNbWHHyorRERIw5STMX0JB8Xkit84MF20w3NMRQs6qDF6AkJVY/OUFUgupeXXjQaALHcQG74u2F+4bq38uiC387kKMzbA/b5N51ZQkeJipPjo5tMpS+HXU54QEbJAKkmavQhuBgCqLSUlRRxGt2l6f+Ny9AT53/WIMB3V3y9nZlr8glhyItcKEVVBAWd7d252U+6ylymx30REkDpYCHeQ=
   - secure: D1mYdwt1Vz0CQbZJx/87AgmKIGV1qdaDPGABoSz5YchS8NrsFXkcty+WfytpiBfWqx/8mXaT0Y6MUcl888L6gRoGSVjRsWu7l5KhchiRTStW63frm1VkClvJIk5PTZvyH8S7XJNJ7TjqUhtlXk3BsMzFSqor456ssc7rpEI+WIdRmV4VwMuIXe35W/mJs3kXEJXTks2DkUSzjKYD8yOQA7mnsv2J/q0WaPDZGagAD7AMH89/Tveku970UvrbOIIBpzhA4TA04OW/d1pwotod/lZnTDp8N722yGaKeCO+ehw/IolMNLY/ivKWyZxfibP9TUaTOY1h0lwLj2VmixAvlSZdwmfFDO69mbB8tRQ3aNI6bX7gmWTnGcPh2aZ6cnHPrZ43C2jVAP0YOHZ7Rmc+U22wHyfgHsPmP36t2mJrN3AjB8vhcY10g9GuhRQ9AHX5A8Os5jET/7RoC0o03xBCXc+fTngkASyaSmavnIhZK3pUs66fH71GOE0IujtvhtvMJFOWn5mxEWUURfyZ+X/bRoG5+Q+sFgSH41humJEc+PDfj3AiyVtmYYve1QD1xeKW1UDAtCkhV0gZCeGrhloDRaWUDFh3svRSJdW6R8+QJf327NnLk2kBVLuKQTOUGgTeo+rLbk6EKc3kWBCxG8iYX6LBqDUIiQyxA9YwpjWYnAo=
-script:
-  #Docker login
-- echo $DockerPassword | docker login --username joefish29 --password-stdin
 
-#Test
-- docker build --target test --tag my-test-image .
-- docker run --env-file env_test my-test-image test_client.py
-- docker run --env-file env_test my-test-image test_items.py
-- docker run -e Key=$Key -e TOKEN=$TOKEN -e BoardID=$BoardID -e ToDoId=$ToDoId -e
-  DoingId=$DoingId -e DoneId=$DoneId my-test-image test_system.py
+stages:
+  - test
+  - name: deploy
+    if : branch = master
 
-  #Heroku
-- echo $HEROKU_API_KEY | docker login --username=_ --password-stdin registry.heroku.com
-- docker pull joefish29/todo-app
-- docker tag joefish29/todo-app registry.heroku.com/todoapp-joefish/web
-- docker push registry.heroku.com/todoapp-joefish/web
-- heroku container:release web -a todoapp-joefish
+jobs:
+  include:
+    - stage: test
+      script:
+        #Docker login
+        - echo $DockerPassword | docker login --username joefish29 --password-stdin
 
-#DockerHub
-- docker build --target production --tag joefish29/todo-app:latest .
-- docker push joefish29/todo-app:latest
+        #Test
+        - docker build --target test --tag my-test-image .
+        - docker run --env-file env_test my-test-image test_client.py
+        - docker run --env-file env_test my-test-image test_items.py
+        - docker run -e Key=$Key -e TOKEN=$TOKEN -e BoardID=$BoardID -e ToDoId=$ToDoId -e
+          DoingId=$DoingId -e DoneId=$DoneId my-test-image test_system.py
+
+    - stage: deploy
```

Comment:
Doing this as a stage is fine, but there is also an existing deploy step (rather than just a named stage) that may make this easier:
[Docs here](https://docs.travis-ci.com/user/deployment) and note that this explicitly says that it skips the deployment stages on pull requests